Mathlib/Data/List/Basic.lean

```diff
@@ -2209,6 +2209,18 @@ theorem modifyNthTail_eq_take_drop (f : List α → List α) (H : f [] = []) :
   | n + 1, b :: l => congr_arg (cons b) (modifyNthTail_eq_take_drop f H n l)
 #align list.modify_nth_tail_eq_take_drop List.modifyNthTail_eq_take_drop
 
+@[simp]
+theorem modifyNth_nil (f : α → α) (n) :
+    modifyNth f n [] = [] := by cases n <;> rfl
+
+@[simp]
+theorem modifyNth_zero_cons (f : α → α) (a l) :
+    modifyNth f 0 (a :: l) = f a :: l := rfl
+
+@[simp]
+theorem modifyNth_succ_cons (f : α → α) (n a l) :
+    modifyNth f (n.succ) (a :: l) = a :: modifyNth f n l := rfl
```

Comment:
```suggestion
    modifyNth f n.succ (a :: l) = a :: modifyNth f n l := rfl
```
or should we use `n + 1`?

Reply:
Yes, ok the surrounding text uses n+1, so I'll change that